posthog-android/src/main/java/com/posthog/android/replay/PostHogReplayIntegration.kt

```diff
@@ -607,43 +607,59 @@ public class PostHogReplayIntegration(
+            try {
+                val semanticsOwner =
+                    (view as? RootForTest)?.semanticsOwner ?: run {
+                        config.logger.log("View is not a RootForTest: $view")
+                        return@post
+                    }
+                val semanticsNodes = semanticsOwner.getAllSemanticsNodes(true)
 
-                val hasMaskModifier = node.config.contains(PostHogReplayMask)
-                val isNoCapture = hasMaskModifier && node.config[PostHogReplayMask]
+                semanticsNodes.forEach { node ->
+                    val hasText = node.config.contains(SemanticsProperties.Text)
+                    val hasEditableText = node.config.contains(SemanticsProperties.EditableText)
+                    val hasPassword = node.config.contains(SemanticsProperties.Password)
+                    val hasImage = node.config.contains(SemanticsProperties.ContentDescription)
 
-                when {
-                    isNoCapture -> {
-                        maskableWidgets.add(node.boundsInWindow.toRect())
-                    }
+                    val hasMaskModifier = node.config.contains(PostHogReplayMask)
+                    val isNoCapture = hasMaskModifier && node.config[PostHogReplayMask]
 
-                    !hasMaskModifier -> {
-                        when {
-                            (hasText || hasEditableText) && (config.sessionReplayConfig.maskAllTextInputs || hasPassword) -> {
-                                maskableWidgets.add(node.boundsInWindow.toRect())
-                            }
+                    when {
+                        isNoCapture -> {
+                            maskableWidgets.add(node.boundsInWindow.toRect())
+                        }
 
-                            hasImage && config.sessionReplayConfig.maskAllImages -> {
-                                maskableWidgets.add(node.boundsInWindow.toRect())
+                        !hasMaskModifier -> {
+                            when {
+                                (hasText || hasEditableText) && (config.sessionReplayConfig.maskAllTextInputs || hasPassword) -> {
+                                    maskableWidgets.add(node.boundsInWindow.toRect())
+                                }
+
+                                hasImage && config.sessionReplayConfig.maskAllImages -> {
+                                    maskableWidgets.add(node.boundsInWindow.toRect())
+                                }
                             }
                         }
                     }
                 }
+            } catch (e: Throwable) {
+                // swallow possible errors due to compose versioning, etc
+                config.logger.log("Session Replay findMaskableComposeWidgets (main thread) failed: $e")
+            } finally {
+                latch.countDown()
             }
+        }
+
+        try {
+            // await for 1s max
+            latch.await(1000, TimeUnit.MILLISECONDS)
```

Comment:
just an arbitrary value since it is a background thread, awaiting time wont block the screen, the time-sensitive is the part within the main thread block